### ISSUE TITLE
TypeCategory::from does not handle Utf8View

### ISSUE THREAD

**Issue by Omega359** [apache/datafusion#13358]
### Describe the bug

Utf8View currently is matching as Unsupported but should be matched as unknown in [TypeCategory::from](https://github.com/apache/datafusion/blob/be19afca4ca93cf7ae0305aafec10b29a8140a58/datafusion/expr-common/src/type_coercion/binary.rs#L307)

```
// String literal is possible to cast to many other types like numeric or datetime,
// therefore, it is categorized as an unknown type
if matches!(
    data_type,
    DataType::Utf8 | LargeUtf8 | DataType::Null
) {
    return TypeCategory::Unknown;
}
```

This should include Utf8View 

### To Reproduce

See code.

### Expected behavior

Utf8View should map to TypeCategory::Unknown

**Comment by Omega359** [comment]
I think this is taken care of by https://github.com/apache/datafusion/pull/13350

**Comment by Omega359** [comment]
Fixed by #13350
